tests/GenFu.Tests/When_testing_for_default_values.cs

```diff
@@ -15,112 +15,127 @@ public class When_testing_for_default_values
         {
             var post = new BlogPost { BlogPostId = 12 };
-            Assert.True(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == "BlogPostId")));
+            Assert.True(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == nameof(BlogPost.BlogPostId))));
         }
 
         [Fact]
         public void Default_string_should_return_false()
         {
             var post = new BlogPost { };
-            Assert.False(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == "Title")));
+            Assert.False(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == nameof(BlogPost.Title))));
         }
 
         [Fact]
         public void Filled_string_should_return_true()
         {
             var post = new BlogPost { Title = "Bob" };
-            Assert.True(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == "Title")));
+            Assert.True(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == nameof(BlogPost.Title))));
         }
 
         [Fact]
         public void Default_datetime_should_return_false()
         {
             var post = new BlogPost { };
-            Assert.False(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetRuntimeProperties().First(x => x.Name == "CreateDate")));
+            Assert.False(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetRuntimeProperties().First(x => x.Name == nameof(BlogPost.CreateDate))));
         }
 
         [Fact]
         public void Filled_datetime_should_return_true()
         {
             var post = new BlogPost { CreateDate = DateTime.Now };
-            Assert.True(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == "CreateDate")));
+            Assert.True(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == nameof(BlogPost.CreateDate))));
         }
 
         [Fact]
         public void Filled_enum_should_return_true()
         {
             var post = new BlogPost { Type = BlogTypeEnum.Post};
-            Assert.True(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == "Type")));
+            Assert.True(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == nameof(BlogPost.Type))));
         }
 
         [Fact]
         public void Empty_enum_should_return_false()
         {
             var post = new BlogPost();
-            Assert.False(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == "Type")));
+            Assert.False(DefaultValueChecker.HasValue(post, typeof(BlogPost).GetProperties().First(x => x.Name == nameof(BlogPost.Type))));
+        }
+
+        [Fact]
+        public void Empty_collections_should_return_false()
```

Comment:
Is there a test somewhere that checks null collections?

Reply:
There is now.

Reply:
cool